src/WebJobs.Extensions.Http/ClaimsIdentityBindingProvider.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsIdentity"/>.
+    /// </summary>
+    /// <remarks>
+    /// Attempts to bind to an identity in precedence order based on <see cref="AuthorizationLevel"/> levels.
+    /// I.e. if the request is key authenticated (Function/System/Admin) that identity is returned, if the
+    /// request is user authenticated (User) that identity is returned.
+    /// </remarks>
+    internal class ClaimsIdentityBindingProvider : IBindingProvider
+    {
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsIdentity))
+            {
+                return Task.FromResult<IBinding>(null);
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public async Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                // if the request is EasyAuth authenticated, get the identity provider type
+                object value = null;
+                string identityProvider = null;
+                if (context.BindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out value))
+                {
+                    IDictionary<string, string> headers = (IDictionary<string, string>)value;
+                    if (headers != null)
+                    {
+                        headers.TryGetValue(HttpConstants.AntaresEasyAuthProviderHeaderName, out identityProvider);
+                    }  
+                }
+
+                var identity = GetPrimaryIdentity(ClaimsPrincipal.Current, identityProvider);
```

Comment:
`ClaimsPrincipal.Current` won't work in Functions v2. I suggest we avoid the dependency on `System.Security.Claims` entirely.

Reply:
Yeah, this was mentioned in the previous code review. I haven't removed the Claims based Identity providers yet from Matthews original work.

---
In reply to: [171723801](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/373#discussion_r171723801) [](ancestors = 171723801)